src/main/java/com/github/princesslana/eriscasper/data/util/AuditLogChangeDeserializer.java

```diff
@@ -0,0 +1,165 @@
+package com.github.princesslana.eriscasper.data.util;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.github.princesslana.eriscasper.data.Data;
+import com.github.princesslana.eriscasper.data.DataException;
+import com.github.princesslana.eriscasper.data.Snowflake;
+import com.github.princesslana.eriscasper.data.resource.AuditLogChange;
+import com.github.princesslana.eriscasper.data.resource.ImmutableAuditLogChange;
+import com.github.princesslana.eriscasper.data.resource.Overwrite;
+import com.github.princesslana.eriscasper.data.resource.Role;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.util.Optional;
+
+public class AuditLogChangeDeserializer extends StdDeserializer<AuditLogChange> {
+
+  AuditLogChangeDeserializer() {
```

Comment:
why package protected?

Reply:
There's no need for it to be implemented outside of the jackson mapper we setup with Data. Data should always be the only one touching Audit-Logs, I can move it into a public setting though.

Reply:
yeah, I'd prefer public